src/pages/atomicdex/api/v20-dev/non_fungible_tokens/index.mdx

```diff
@@ -0,0 +1,594 @@
+| to                   | array            | List of destination addresses.                                                                   |
+| contract_type        | string           | Contract type. `ERC721` or `ERC1155`.                                                            |
+| token_address        | string           | Token address.                                                                                   |
+| token_id             | string           | Token ID.                                                                                        |
+| fee_details          | object           | A standard [WithdrawFee](/atomicdex/api/v20#withdraw-fee) object.                                |
+| coin                 | string           | Coin name. One of `AVALANCHE`, `BSC`, `ETH`, `FANTOM`, or `POLYGON`.                             |
+| block_height         | integer          | Block height of the transaction. If the value is `0`, the transaction is unconfirmed.            |
+| timestamp            | integer          | Timestamp of the block containing the withdrawl transaction.                                     |
+| internal_id          | integer          | Defaults to `0`, but may be used in future for backend referencing to identify a transaction in local database data. |
+| transaction_type     | string           | This will always be `NftTransfer`.                                                               |
+
+<DevComment>
+Is `transaction_type` always `NftTransfer`?
+</DevComment>
+
+<CollapsibleSection expandedText='Hide Response' collapsedText='Show Response'>
+```json
+{
+  "mmrpc": "2.0",
+  "result": {
+    "tx_hex": "f8ca0c84b2d05e008301acab94fd913a305d70a60aac4faac70c739563738e1f8180b86442842e0e000000000000000000000000f622a6c52c94b500542e2ae6bcad24c53bc5b6a20000000000000000000000006fad0ec6bb76914b2a2a800686acc2297064582000000000000000000000000000000000000000000000000000000031e5472c7e8194a07eaa06fe70087b8fef9774472c3bc72ad4c6814ea1fac6515a45987864892f0ca05806e069db608f26f7442595815fbe04f117f10171557e7228d354d0e2f2ac05",
+    "tx_hash": "5d93549408b3bae1c7a03fcc37a7cb7cca37b141de9119c2b87cff1d540507d7",
+    "from": [
+      "0xf622a6C52C94b500542E2AE6bcAD24C53Bc5b6a2"
+    ],
+    "to": [
+      "0x6FAD0eC6bb76914b2a2a800686acc22970645820"
+    ],
+    "contract_type": "ERC721",
+    "token_address": "0xfd913a305d70a60aac4faac70c739563738e1f81",
+    "token_id": "214300044414",
+    "amount": "1",
+    "fee_details": {
+      "type": "Eth",
+      "coin": "BNB",
+      "gas": 109739,
+      "gas_price": "0.000000003",
+      "total_fee": "0.000329217"
+    },
+    "coin": "BNB",
+    "block_height": 0,
+    "timestamp": 1684239228,
+    "internal_id": 0,
+    "transaction_type": "NftTransfer"
+  },
+  "id": null
+}
+```
+</CollapsibleSection>
+
+
+#### 📌 ERC1155 Withdraw Example
+
+If you are sending 2 or more NFTs, you must use the `withdraw_erc1155` withdraw type.
```

Comment:
this sentence has wrong meaning. withdraw_types `erc1155` or `erc721` dont depend on how many tokens you want to withdraw. 
it depends on `contract type/contract standard/token standard`. NFTs which we support have a solid `contract_type`: `erc721` or `erc1155`.
So what I want to say, user must use withdraw_erc721 if they want to withdraw nft which has erc721 contract_type.
If nft has erc1155 standard, then user must use withdraw_erc1155.

If nft has "contract_type": "ERC721" it means that it is absolutely unique . It has only 1 owner and owner can own only 1 NFT of some token_id in some token_address (token address actually is a contract address).

If nft has "contract_type": "ERC1155", it means that this standard allows more than 1 users to own more than 1 nft with identical token_id. 

So withdraw_type in request form have different fields for two different token standards.

```rust
// this struct is actually what you build for nft withdraw request
#[derive(Clone, Deserialize)]
pub struct WithdrawNftReq {
    pub(crate) url: Url,
    pub(crate) withdraw_type: WithdrawNftType,
}

#[derive(Clone, Deserialize)]
#[serde(tag = "type", content = "withdraw_data")]
#[serde(rename_all = "snake_case")]
pub enum WithdrawNftType {
    WithdrawErc1155(WithdrawErc1155),
    WithdrawErc721(WithdrawErc721),
}

#[derive(Clone, Deserialize)]
pub struct WithdrawErc1155 {
    pub(crate) chain: Chain,
    pub(crate) to: String,
    pub(crate) token_address: String,
    pub(crate) token_id: BigDecimal,
    pub(crate) amount: Option<BigDecimal>,
    #[serde(default)]
    pub(crate) max: bool,
    pub(crate) fee: Option<WithdrawFee>,
}

#[derive(Clone, Deserialize)]
pub struct WithdrawErc721 {
    pub(crate) chain: Chain,
    pub(crate) to: String,
    pub(crate) token_address: String,
    pub(crate) token_id: BigDecimal,
    pub(crate) fee: Option<WithdrawFee>,
}
```

As you can see in `withdraw_type` field we dont have one struct, we have enum type which contains two enumerations, [not WithdrawData object](https://github.com/KomodoPlatform/komodo-docs-mdx/pull/42/files#diff-965f9228b14137d6315455493012e3ebadd2db27aa3428da3e7cec792ef8270aR330-R332). 

So for instance "type": "withdraw_erc721" means that it is enumeration `WithdrawErc721` from `enum WithdrawNftType` and `withdraw_data` contains field for WithdrawErc721 
```rust
pub struct WithdrawErc721 {
    pub(crate) chain: Chain,
    pub(crate) to: String,
    pub(crate) token_address: String,
    pub(crate) token_id: BigDecimal,
    pub(crate) fee: Option<WithdrawFee>,
}
```
we just dont specify name of `WithdrawErc721` in json, bcz back end will build itself the object from fields .
```json
"withdraw_type": {
      "type": "withdraw_erc721",
      "withdraw_data": {
        "chain": "BSC",
        "to": "0x6FAD0eC6bb76914b2a2a800686acc22970645820",
        "token_address": "0xfd913a305d70a60aac4faac70c739563738e1f81",
        "token_id": "214300044414"
      }
    }
```

So [here](https://github.com/KomodoPlatform/komodo-docs-mdx/pull/42/files#diff-965f9228b14137d6315455493012e3ebadd2db27aa3428da3e7cec792ef8270aR330-R343) you need to provide two variants of fields.
If contact type erc721 then: `chain, to, token_address, token_id, fee.`
If contact type erc1155 then:  `chain, to, token_address, token_id, amount, max, fee`

Reply:
thanks for clarification, I see the last description was a bit ambiguous.

Reply:
I've created two standard objects to reference this